packages/flame/lib/src/game/mixins/single_game_instance.dart

```diff
@@ -0,0 +1,29 @@
+import '../../components/component.dart';
+import 'game.dart';
+
+/// Mixin that declares a [Game] class as a singleton.
+///
+/// In many applications there is a single [Game] instance within the entire
+/// app. For example, most regular game apps fall within this category. If your
+/// application is structured like that, you can use this mixin to declare your
+/// game class as a "singleton game".
+///
+/// The effect of using this mixin is that it allows component lifecycle events
+/// to be resolved faster in certain situations. For example, [Component]'s
+/// `onLoad` will be able to run even if its parent is currently detached.
+///
+/// Using this mixin in applications where there are multiple [Game] instances
+/// attached to multiple GameWidgets, is invalid.
+mixin SingleGameInstance on Game {
+  @override
+  void onMount() {
+    Component.staticGameInstance = this;
```

Comment:
we could add some checks like: `if staticGameInstance != null && != this` then throw pretty error message saying you cannot use SingleGameInstance like that

Reply:
There is an assert like that in `FlameGame` class:
```dart
assert(
      Component.staticGameInstance == null,
      '$this instantiated, while another game ${Component.staticGameInstance} '
      'declares itself to be a singleton',
    );
```
I think it's better to have it in FlameGame because then not only we protect against two `SingleGameInstance` objects, but we also prevent a `SingleGameInstance` to coexist with a regular `FlameGame`.